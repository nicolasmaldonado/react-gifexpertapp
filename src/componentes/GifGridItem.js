

export const GifGridItem = ( gif ) => {

  return (
    <div className='card animate__animated animate__fadeIn'>
      <img src={ gif.url } alt={ gif.title }/>
      <p>{ gif.title }</p>
    </div>
  )
}