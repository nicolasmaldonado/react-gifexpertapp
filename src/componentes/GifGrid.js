import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";


export const GifGrid = ({ category }) => {

  const { data: gifs, loading } = useFetchGifs( category );

  return (
    <>
      <h3>{ category }</h3>

      { loading && <p className='animate__flash'>Loading</p> }

      <div className='card-grid'>

        {
          gifs.map( (gif) => {
            return ( 
              <GifGridItem 
                key={ gif.id }
                { ...gif }
              />
            )
          })
        }

      </div>
    </>
  )
}