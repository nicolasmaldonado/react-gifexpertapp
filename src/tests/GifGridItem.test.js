import { shallow } from "enzyme";
import { GifGridItem } from "../componentes/GifGridItem";

describe('Testing GifGridItem', () => {

  let wrapper;
  const title = 'New title';
  const url = 'https://google.com';

  beforeEach( () => {
    wrapper = shallow(
      <GifGridItem 
        title = { title }
        url = { url }
      />
    );
  })

  test('Debe mostrar el componente correctamente', () => {
    expect( wrapper ).toMatchSnapshot();
  })
  
  test('Debe tener un parrafo con el title', () => {
    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe( title );
  })

  test('Debe obtener la imagen con matching url y alt', () => {
    const img = wrapper.find('img');
    expect( img.props().src ).toBe( url );
    expect( img.props().alt ).toBe( title )
  })

  test('Debe tener animate__fadeIn', () => {
    const div = wrapper.find('div');
    //expect( div.prop('className') ).toMatch('animate__fadeIn');
    expect( div.hasClass('animate__fadeIn') ).toBeTruthy();
  })
  
  

})
