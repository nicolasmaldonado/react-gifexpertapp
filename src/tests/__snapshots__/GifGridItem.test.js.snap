// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="New title"
    src="https://google.com"
  />
  <p>
    New title
  </p>
</div>
`;
